### ISSUE TITLE
training and validation data is the same, but loss and metrics show very different values 

### ISSUE THREAD

**Issue by vahe4060** [qubvel/segmentation_models#483]
I want to use a Unet model from segmentation_models library for a binary classification task. However I don't know why my model behaves very 'strange'. Here is the code for my model.

```
import segmentation_models as sm
sm.set_framework('tf.keras')

BACKBONE = 'resnet34'
preprocess_input = sm.get_preprocessing(BACKBONE)

# preprocess input
x_train = preprocess_input(train_x)
    
model = sm.Unet(BACKBONE, encoder_weights='imagenet')
model.compile('Adam', loss=sm.losses.bce_jaccard_loss, metrics=[sm.metrics.iou_score])

history = model.fit(x_train, train_y, 10, epochs=20, validation_data=(x_train, train_y)
```

The strange thing is that the model doesn't seem to keep the updated weights after each epoch and gives horrendous loss and accuracy on any data used for train.

> 
> Epoch 1/20 8/8 [===========] - 6s 300ms/step - loss: 1.4448 - iou_score: 0.1352 - val_loss: 3.5120 - val_iou_score: 0.1660
> 
> Epoch 2/20 8/8 [===========] - 2s 193ms/step - loss: 1.1791 - iou_score: 0.2088 - val_loss: 10.717 - val_iou_score: 0.1657
> 
> Epoch 3/20 8/8 [===========] - 2s 195ms/step - loss: 1.0651 - iou_score: 0.2420 - val_loss: 3.6475 - val_iou_score: 0.0106
> 
> Epoch 4/20 8/8 [===========] - 2s 196ms/step - loss: 0.9659 - iou_score: 0.3050 - val_loss: 3.5878 - val_iou_score: 0.0045
> 
> Epoch 5/20 8/8 [===========] - 2s 197ms/step - loss: 0.9508 - iou_score: 0.3179 - val_loss: 3.5400 - val_iou_score: 0.0280
> 
> Epoch 6/20 8/8 [===========] - 2s 203ms/step - loss: 0.8339 - iou_score: 0.3836 - val_loss: 3.5804 - val_iou_score: 4.69e-06

Would be grateful for any suggestions.

**Comment by elliestath** [comment]
Hey, any update on this? Did you find the problem?

**Comment by vahe4060** [comment]
make sure x_train, y_train are uint8 from {0..255}, and x_train is  preprocessed for the backbone you choose.

**Comment by vahe4060** [comment]
> Hey, any update on this? Did you find the problem?

make sure x_train, y_train are uint8 from {0..255}, and x_train is preprocessed for the backbone you choose.
